### ISSUE TITLE
CT data preprocessing

### ISSUE THREAD

**Issue by ZetrextJG** [reginabarzilaygroup/Sybil#74]
Hi, thanks for the awesome work on the paper. 

I am currently exploring the capabilities of your pretrained model on an external LDCT dataset and have noticed a discrepancy in the preprocessing steps described  in the Paper's Supplementary material and the code available on the `main` branch. 

From the Supplementary Methods section about the Image pre-processing I read:
> To ensure that volumes represented a consistent three-dimensional physical space, series were resampled to a fixed 0.7x0.7x2.5 mm3 voxels and either truncated or padded to obtain 200 slices. Before feeding an exam into the model, the volume was resized to a height and width of 256-by-256 pixels.

However In the code for Series I can see a resample transform (to 0.7x0.7x2.5) followed by a CenterCropPad to (256, 256, 200).

Which of the following methods were used to train the model available  for download under the  Google drive link (https://drive.google.com/drive/folders/1nBp05VV9mf5CfEO6W5RY4ZpcpxmPDEeR)?

Also the cropping step seems weird considering the fact that the attention visualization methods interpolate the resulting maps to the entire 512x512xD image and not the cropped 256x256x200 part.

**Comment by pgmikhael** [comment]
Hi,

Thanks for reaching out. 

For the risk prediction, we train and test on images first scaled (each slice independently) to the 256x256 size. In the Serie object, the `self._loader` is what reads the images (executed [here](https://github.com/reginabarzilaygroup/Sybil/blob/main/sybil/loaders/abstract_loader.py#L159)). This passes the individual slices through a sequence of augmentations, one of which is the resizing (see [here](https://github.com/reginabarzilaygroup/Sybil/blob/main/sybil/utils/loading.py#L161)). Then the resampling and cropping is done. This is all consistent with how the model was trained and as we tried to describe in the methods. 

Regarding the attention visualization, the size doesn't really matter. Attention scores are computed on the latent features of the model before pooling. For visualizing purposes, we use the interpolation function to resize to the input image size. You can go back to the training image size or original image size of 512x512 -- both are coarse visualizations.

Hope this helps.

**Comment by ZetrextJG** [comment]
Thank you very much for a swift response.

Is the spacing information contained in the dicom unchanged when the initial resizing  to 256x256 is done?
This would result in doubling the spacing on those two axis implicitly and result in the image not with (0.7x0.7x2.5) but rather (1.4x1.4x2.5) image which is later cropped (or padded) to an volume of 256x256x200. This would make intuitive sense for me.

My real question is: Is the procedure
1. Load that volume (in my case it is stored as nibabel image)
2. Resample the volume to (1.4x1.4x2.5)
3. CenterCropPad to (256x256x200)
4. Feed it to the model with shape (1, 1, 200 , 256, 256)

analogous to the original?

**Comment by ZetrextJG** [comment]
I have examined the preprocessing code multiple times now and compared it to the procedure described by me above and it seems to align perfectly. With exactly `1.40625` instead approximate 1.4 I put in the comment above. 

Could You @pgmikhael confirm this?

**Comment by pgmikhael** [comment]
Hi,

Yes, aside from the normalization, this should be right.

**Comment by ZetrextJG** [comment]
Of course, windowing and normalization applied. 

Thank you very much. I am considering this issue resolved.
